src/Microsoft.ComponentDetection.Detectors/pip/PythonResolver.cs

```diff
@@ -165,7 +169,24 @@ private async Task<IList<PipGraphNode>> ProcessQueueAsync(ISingleFileComponentRe
 
         node.Value = new PipComponent(pipComponent.Name, candidateVersion, license: pipComponent.License, author: pipComponent.Author);
 
-        var dependencies = (await this.FetchPackageDependenciesAsync(state, newSpec)).ToDictionary(x => x.Name, x => x);
+        var fetchedDependences = await this.FetchPackageDependenciesAsync(state, newSpec);
+
+        // Multiple dependency specification versions can be given for a single package name
+        // Until a better method is divised, choose the latest entry
+        var dependencies = new Dictionary<string, PipDependencySpecification>();
+        fetchedDependences.ForEach(d =>
```

Comment:
Do we have a pattern for using an id rather than a name in other detectors?

Reply:
I think I'm comfortable if you're planning to handle this in a future PR--these dependencies aren't getting registered today because of the exception so there's not a risk of dropping deps.

Reply:
Yes, plan is to handle them in #963